cordformation/src/main/kotlin/net/corda/plugins/Dockerform.kt

```diff
@@ -146,33 +144,70 @@ open class Dockerform @Inject constructor(objects: ObjectFactory) : Baseform(obj
 
                 it.installDefaultDatabaseConfig(dbConfig)
 
-                val args = mutableMapOf<String, Any>()
-
-                dockerfileArgs.entrySet().toList().map {
-                    args[it.key] = it.value.unwrapped()
-                }
-
-                services[dbHost] = mapOf(
+                val dbDockerfile = dockerConfig.getString("dockerConfig.dbDockerfile")
+                // Override the port and hostname parameters in dockerfile
+                val dbDockerfileArgs = defaultUrlArgs.withFallback(
+                        dockerConfig.getConfig("dockerConfig.dbDockerfileArgs")).resolve()
+                val database = mutableMapOf(
                         "build" to mapOf(
                              "context" to project.buildDir.absolutePath,
                              "dockerfile" to project.buildDir.resolve(dbDockerfile).toString(),
-                             "args" to args
+                             "args" to dbDockerfileArgs.entrySet().map { it.key to it.value.unwrapped() }.toMap()
```

Comment:
Ouch! :face_with_head_bandage:. How about:
```kotlin
dbDockerfileArgs.entrySet().associate { it.key to it.value.unwrapped() }
```
Kotlin is full of inline lambdas like these :man_facepalming:. Just gotta learn 'em all...